engine/XcodeUnitTestEngine.php

```diff
@@ -50,7 +51,9 @@ private function shouldGenerateCoverage() {
     //        available.
     // true:  User passed --coverage.
     // https://secure.phabricator.com/T10561
-    return $this->getEnableCoverage() !== false;
+    $arcCoverageFlag = $this->getEnableCoverage();
+
+    return ($arcCoverageFlag !== false) && $this->hasCoverageKey;
```

Comment:
Just add the extra check to make sure the `coverage` key exists before telling Xcode to run with coverage enabled.